### PR TITLE
In HOWTO-make-a-release.md, include what's needed for premium releases

### PR DESCRIPTION
**Pull request by levitte**
Premium releases aren't terribly different from public releases, it's all in the details.

There were a couple of other details that had aged and weren't current any more, they are fixed too.

### CODE REVIEW

File: HOWTO-make-a-release.md [hunk 1]
```diff
@@ -137,19 +143,27 @@ You will need to checkout at least three working trees:
 
 ## Freeze the source repository
 
-Three business day before the release, freeze the main repository.  This
-locks out everyone but the named user, who is doing the release, from doing
-any pushes.  Someone other than the person doing the release should run the
-command.
+Three business day before the release, freeze the appropriate source
+repository.
```
Comment (mattcaswell) [inline HOWTO-make-a-release.md:147]: At the moment we only freeze the main repo. Is it possible to freeze the premium repo?

File: HOWTO-make-a-release.md [hunk 2]
```diff
@@ -243,20 +259,27 @@ The manual for that script is found in `$TOOLS/release-tools/MKRELEASE.md`
...
-*Do* send the commits to the reviewer and await their approval.
+    If there is a Security Advisory then copy it into the secadv directory.
 
 Commit your changes, but *do not push* them to the release data repo at this
 stage.  (the release data repo being `git@github.openssl.org:omc/data.git`)
 
+*Do* send the commits to the reviewer and await their approval.  It's
+advisable to use this command to get a copy of those commits:
+
+    git format-patch
```
Comment (mattcaswell) [inline HOWTO-make-a-release.md:281]: Now this is in the data repo, none of this seems necessary any more. Just create a PR.

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: HOWTO-make-a-release.md (hunk 1 above)
Comment (levitte) [reply HOWTO-make-a-release.md:147]: Yes.  I checked, the frozen hook is enabled

File: HOWTO-make-a-release.md (hunk 2 above)
Comment (levitte) [reply HOWTO-make-a-release.md:281]: Oh!  Right, this was from gitolite times...

File: HOWTO-make-a-release.md [hunk 3]
```diff
@@ -208,8 +223,9 @@ match in the .md5, .sha1, .sha256, and review the announcment file.
 *Do* send the auto-generated commits to the reviewer and await their
 approval.
 
-*Do not push* changes to the main source repo at this stage.
-(the main source repo being `git@github.openssl.org:openssl/openssl.git`)
+*Do not push* changes to the source repo at this stage.
+(the source repo being on of `git@github.openssl.org:openssl/openssl.git` or
+`git@github.openssl.org:openssl/premium.git`)
```
Comment (levitte) [inline HOWTO-make-a-release.md:228]: @mattcaswell, your comment further down (near `git format-patch`) made me wonder if this is true any more, or needs an update as well...

Comment (mattcaswell) [reply HOWTO-make-a-release.md:228]: This is still correct. Pushing the changes to the source repo makes them public (at least for the public repo) and we do that quite late in the process.

File: HOWTO-make-a-release.md [hunk 4]
```diff
@@ -184,8 +198,9 @@ Reviewed-By trailers as required.
...
```
[the remark this answers is not in the document]
Comment (mattcaswell) [reply HOWTO-make-a-release.md:203]: This is still a thing, although I tend to do it slightly differently for a release with security fixes vs one without. If there are no security fixes then CHANGES/NEWS updates and copyright year updates can be pushed straight away (no need to hold them back). If security fixes are included then that affects the CHANGES/NEWS and copyright changes so I keep all of those changes back and don't push them yet.

[the remark this answers is not in the document]
Comment (levitte) [reply HOWTO-make-a-release.md:203]: Should we file that under "Matt does things a little differently but ok" or should we record that in this file as something one can optionally do?

File: HOWTO-make-a-release.md (hunk 1 above)
Comment (t8m) [reply HOWTO-make-a-release.md:147]: Not that we really need to freeze it given the frequency of pushes there and that we push there only on particular circumstances.

**Comment by t8m** [comment]
What about instead of pulling the list of repos from some .yml file we would list all repositories under openssl organization in GHE with some particular topic (like `premium`) set?

**Comment by t8m** [comment]
`gh repo list --topic premium` would do that.

**Comment by levitte** [comment]
> What about instead of pulling the list of repos from some .yml file we would list all repositories under openssl organization in GHE with some particular topic (like `premium`) set?

That discussion doesn't really belong here, does it?

File: HOWTO-make-a-release.md (hunk 2 above)
Comment (levitte) [reply HOWTO-make-a-release.md:281]: Rewritten

File: HOWTO-make-a-release.md (hunk 1 above)
Comment (levitte) [reply HOWTO-make-a-release.md:147]: So.....  should I modify this, or leave it as is?

**Comment by levitte** [comment]
I've addressed most review comments, but one is still left with a question.

File: HOWTO-make-a-release.md (hunk 1 above)
Comment (mattcaswell) [reply HOWTO-make-a-release.md:147]: We've never previously frozen it, so I think lets not bother doing this for premium. It seems over the top.

File: HOWTO-make-a-release.md [hunk 5]
```diff
@@ -184,8 +198,9 @@ Reviewed-By trailers as required.
 *Do* send the auto-generated commits to the reviewer and await their
 approval.
 
-*Do not push* changes to the main source repo at this stage.
-(the main source repo being `git@github.openssl.org:openssl/openssl.git`)
+*Do not push* changes to the source repo at this stage.
+(the source repo being on of `git@github.openssl.org:openssl/openssl.git` or
```
Comment (mattcaswell) [inline HOWTO-make-a-release.md:202]: typo: one of

File: HOWTO-make-a-release.md [hunk 6]
```diff
@@ -208,8 +223,9 @@ match in the .md5, .sha1, .sha256, and review the announcment file.
...
```
Comment (mattcaswell) [inline HOWTO-make-a-release.md:227]: typo: one of

File: HOWTO-make-a-release.md [hunk 7]
```diff
@@ -282,44 +303,49 @@ The do-release.pl script will display the commands you will need to issue to
...
 
     git push <repository> <tagname>
 
 ## Updating the release data
 
-Push the newsflash changes to the release data repo.  When you do this, the
-website will get updated and a script to flush the Akamai CDN cache will be
-run.  You can look at things on www-origin.openssl.org; the CDN-hosted
-www.openssl.org should only be a few minutes delayed.
+If there is a PR against the release repo to be merged, perform the merge
```
Comment (mattcaswell) [inline HOWTO-make-a-release.md:327]: Previously this referred to the "release data repo"...the word "data" has been dropped which kind of seems important. When I first read this I was temporarily confused.....what "release repo" are we talking about here?

File: HOWTO-make-a-release.md (hunk 5 above)
Comment (levitte) [reply HOWTO-make-a-release.md:202]: Yup

File: HOWTO-make-a-release.md (hunk 6 above)
Comment (levitte) [reply HOWTO-make-a-release.md:227]: Yup

File: HOWTO-make-a-release.md (hunk 7 above)
Comment (levitte) [reply HOWTO-make-a-release.md:327]: Fixed

**Review by mattcaswell — APPROVED** [review]
LGTM

**Comment by levitte** [comment]
Merged

0cdf6fcb3f958e1f71b2104f60bf1cb632924570 Updates of things that have aged
9cb7b2461030c1542ac2a6a1f039020427a1c250 Cover the premium release as well

**Pull request closed without merging**